vms/platformvm/state/state.go

```diff
@@ -2384,6 +2735,92 @@ func writePendingDiff(
 	return nil
 }
 
+func (s *state) writeSubnetOnlyValidators() error {
+	// Write modified weights
+	for subnetID, weight := range s.sovDiff.modifiedTotalWeight {
+		var err error
+		if weight == 0 {
+			err = s.weightsDB.Delete(subnetID[:])
```

Comment:
What's the expected behavior of `WeightOfSubnetOnlyValidators` after the entry is deleted from the DB and the cache entry expires? If it's to error, perhaps we should not write to the cache so that `WeightOfSubnetOnlyValidators` deterministically errors for any subnets with weight 0.

Reply:
`WeightOfSubnetOnlyValidators` calls `database.WithDefault(database.GetUInt64, s.weightsDB, subnetID[:], 0)` which will return `0` if the value is not on disk. So, `WeightOfSubnetOnlyValidators` deterministically returns `0` for any subnets with weight `0`.